packages/suite-web/e2e/tests/coinmarket/buy.test.ts

```diff
@@ -0,0 +1,191 @@
+        cy.discoveryShouldFinish();
+        // navigate to buy
+        cy.getTestElement('@suite/menu/wallet-index').click();
+        cy.getTestElement('@wallet/menu/wallet-coinmarket-buy').click();
+    });
+
+    /**
+     * 1. Navigates to Trade/Buy.
+     * 2. Verifies the mocked API response (country:AT).
+     * 3. Fills in an amount and clicks “Compare offers”.
+     * 4. Verifies the mocked API response (only offers from the mocked file, e.g. banxa, btcdirect).
+     * 5. Picks one offer and clicks “Get this deal”.
+     * 6. Verifies that a modal opens.
+     * 7. Clicks the checkbox and “Confirm”.
+     * 8. Clicks “Confirm on Trezor”  in Suite and on the emulator.
+     * 9. Verifies “Confirmed on Trezor” text.
+     * 10. Verifies the amount, currency, crypto, provider and payment method all match the mocked/given data.
+     * 11. Clicks “Finish transaction”.
+     * 12. Simulates interaction with the partner's site
+     * 13. Verifies that the buy trade was approved
+     * * 14. Goes back to the Buy tab and verifies the transaction is listed under "Trade transactions"
+     */
+    it.only('Should buy crypto successfully', () => {
+        const testData = {
+            fiatInput: '500',
+            quoteBtcValue: '0.02073954',
+            tradeBtcValue: '0.02066953', // real situation: the final trade value may differ from the quote value
+            btcAddress: 'bc1q7ceqvaq7fqyywxqcx7qnfxkfk2ykpsla9pe80q',
+        };
+
+        // Tests all input windows are empty and country is set to Austria
+        cy.getTestElement('@coinmarket/buy/country-select/input').should('contain.text', 'Austria');
+        cy.getTestElement('@coinmarket/buy/crypto-input').should('have.value', '');
+        cy.getTestElement('@coinmarket/buy/fiat-input').should('have.value', '');
+
+        // Tests set currencies are EUR and BTC
+        cy.getTestElement('@coinmarket/buy/fiat-currency-select/input').should(
+            'contain.text',
+            'EUR',
+        );
+
+        cy.getTestElement('@coinmarket/buy/crypto-currency-select/input').should(
+            'contain.text',
+            'BTC',
+        );
+
+        // Fills out the form
+        cy.getTestElement('@coinmarket/buy/fiat-input').type(testData.fiatInput);
+        cy.getTestElement('@coinmarket/buy/compare-button').click();
+
+        // Verifies the offers displayed match the mock
+        cy.fixture('./invity/buy/quotes').then((quotes: any) => {
+            const exchangeProvider = [
+                ['banxa', 'banxa'],
+                ['btcdirect', 'btcdirect'],
+            ];
+
+            exchangeProvider.forEach((provider: string[]) => {
+                // Tests offer accordance with the mocks
+                const valueFromFixtures = quotes.find(
+                    (quote: any) => quote.exchange === provider[1],
+                );
+                cy.contains('[class*="Quote__Wrapper"]', provider[0], { matchCase: false })
```

Comment:
don't like seeing this. `[class*="Quote__Wrapper"]'` if we, for example, decide to go away from styled components to something else, this might easily change and test break. would be nicer to have data-test here as well